### PR TITLE
[KOGITO-2272] Hot reload test stuck if deployment is not available

### PR DESCRIPTION
**Pull request by danielezonca**
https://issues.redhat.com/browse/KOGITO-2272

### CODE REVIEW

**Comment by danielezonca** [comment]
@evacchi @mariofusco 
Ok I think we have at least a problem in the configuration of the project and maybe a bug in Quarkus too ([logs](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/blue/rest/organizations/jenkins/pipelines/KIE/pipelines/kogito/pipelines/kogito-runtimes/branches/PR-526/runs/1/log/?start=0)):
- I just created a simple test failure in `kogito-codegen` on [purpose](https://github.com/kiegroup/kogito-runtimes/blob/1ac51c72364fc3d97030154757e61af4093b3f1b/kogito-codegen/src/test/java/org/kie/kogito/codegen/RuleUnitCompilerTest.java#L72)
- This makes the build of that module to fail (of course :) )
- All maven dependent module are skipped (in this case `kogito-quarkus-deployment`) [1]
- Hot reload test module doesn't have an explicit dependency declaration on `deployment` so it starts
- `kogito-quarkus-deployment` is not available and it makes the test fails [2]
- The runner stuck until the timeout (this is probably the bug in Quarkus I mentioned) [3]
- Run is aborted and you need to find the _real_ failure in the full log (the last error is completely misleading) [4]

[1]
```
[INFO] ------------------------------------------------------------------------
[INFO] Skipping Kogito - Quarkus - Deployment
[INFO] This project has been banned from the build due to previous failures.
[INFO] ------------------------------------------------------------------------
```
[2]
```
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.kie.kogito:kogito-quarkus-deployment:jar:8.0.0-SNAPSHOT
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:414)
	... 58 more
```
[3]
I do believe this is a bug in Quarkus runner but in all the usages of `QuarkusDevModeTest` I [found](https://github.com/quarkusio/quarkus/search?q=QuarkusDevModeTest&unscoped_q=QuarkusDevModeTest), it is used as integration test _inside_ `deployment` module so they probably do not expect this code to fail. I plan to open an issue on they GitHub later tomorrow.
Is there any reason why we put hot reload test in a different module instead of inside `deployment`? @evacchi @mariofusco 
[4]
```
Timeout has been exceeded
Finished: ABORTED
```

Now I will try to add `kogito-quarkus-deployment` as dependency to Hot reload module with scope `provided` just to inform Maven of this "implicit" dependency and I expect the build to fail properly.

**Comment by danielezonca** [comment]
Ok as expected now the build fails in the proper way :)
My plan now is:
- Revert the broken test
- Make the PR ready for review
- Wait a green build/reviews
- Merge this fix and in parallel show this behavior to Quarkus team

File: kogito-quarkus-extension/integration-test-hot-reload/pom.xml [hunk 1]
```diff
@@ -23,6 +23,15 @@
             <artifactId>drools-decisiontables</artifactId>
         </dependency>
 
+        <!-- this is used implicitly by hot reload tests so let's make Maven aware of it -->
+        <dependency>
+            <groupId>org.kie.kogito</groupId>
+            <artifactId>kogito-quarkus-deployment</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
```
Comment (evacchi) [inline kogito-quarkus-extension/integration-test-hot-reload/pom.xml:31]: i think the scope provided may be unnecessary; if ITs are usually _inside_ the deployment module, then by definition they _do_ depend on the deployment module.

Comment (danielezonca) [reply kogito-quarkus-extension/integration-test-hot-reload/pom.xml]: Done

**Comment by danielezonca** [comment]
@MarianMacik @radtriste 
Can you please review?
The executive summary: the CI build stuck on hotreload test until timeout if there is an error in an upper dependency 
Solution: added a dependency to `deployment` module so Maven is aware of this "implicit dependency" and skip the build of the module if needed

File: kogito-quarkus-extension/integration-test-hot-reload/pom.xml [hunk 2]
```diff
@@ -23,6 +23,14 @@
             <artifactId>drools-decisiontables</artifactId>
         </dependency>
 
+        <!-- this is used implicitly by hot reload tests so let's make Maven aware of it -->
+        <dependency>
+            <groupId>org.kie.kogito</groupId>
+            <artifactId>kogito-quarkus-deployment</artifactId>
+            <version>${project.version}</version>
```
Comment (radtriste) [inline kogito-quarkus-extension/integration-test-hot-reload/pom.xml:30]: could we make it `provided` ?

Comment (danielezonca) [reply kogito-quarkus-extension/integration-test-hot-reload/pom.xml:30]: Lol it was `provided` and then Edoardo proposed to remove `provided` :) ([see comment](https://github.com/kiegroup/kogito-runtimes/pull/526#discussion_r429100522))
Actually I have mixed feeling about this: it is true that provided should be enough (and this is the reason why I did it at the beginning) but it is also true that all other extensions have hot reload tests inside the `deployment` module so it is more similar now with a proper dependency declaration.

Comment (radtriste) [reply kogito-quarkus-extension/integration-test-hot-reload/pom.xml:30]: ahah oki :)
I just read the executive summary before reviewing sry...
Anyway this is just the integration-tests, so it should be ok.

Comment (MarianMacik) [reply kogito-quarkus-extension/integration-test-hot-reload/pom.xml:30]: Another option would be to create an artificial dependency like we have it for KIE Server:

https://github.com/kiegroup/droolsjbpm-integration/blob/9e089567726f5c1fdffa295a81079d45c17aad43/kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/pom.xml#L246-L259

But if the tests depend on it, I would leave it with the default scope as it is now :)

**Pull request merged**